src/jdk.compiler/share/classes/com/sun/tools/javac/code/Types.java

```diff
@@ -4734,6 +4734,10 @@ class Rewriter extends UnaryVisitor<Type> {
 
         boolean high;
         boolean rewriteTypeVars;
+        // map to avoid visiting same type argument twice, like in Foo<T>.Bar<T>
+        Map<Type, Type> argMap = new HashMap<>();
```

Comment:
Do we need to use both a map and a set, or can we just use the map?

Reply:
I think yes and in any case the code is more readable, me thinks, with both than using only one